src/main/java/pl/shop/models/Order.java

```diff
@@ -0,0 +1,62 @@
+package pl.shop.models;
+
+
+import lombok.*;
+
+import javax.persistence.*;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(name = "orders")
+public class Order {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "customer", referencedColumnName = "uuid")
+    private Customer customer;
+
+    private Date ordered;
+
+    private Date shipped;
+
+    @AttributeOverrides({
+            @AttributeOverride(name = "streetName", column = @Column(name = "delivery_address")),
+            @AttributeOverride(name = "streetNumber", column = @Column(name = "delivery_streetNumber")),
+            @AttributeOverride(name = "houseNumber", column = @Column(name = "delivery_houseNumber")),
+            @AttributeOverride(name = "postalCode", column = @Column(name = "delivery_postalCode")),
+            @AttributeOverride(name = "city", column = @Column(name = "delivery_city"))
+    })
+    private Address deliveryAddress;
+
+    @AttributeOverrides({
+            @AttributeOverride(name = "streetName", column = @Column(name = "billing_address")),
+            @AttributeOverride(name = "streetNumber", column = @Column(name = "billing_streetNumber")),
+            @AttributeOverride(name = "houseNumber", column = @Column(name = "billing_houseNumber")),
+            @AttributeOverride(name = "postalCode", column = @Column(name = "billing_postalCode")),
+            @AttributeOverride(name = "city", column = @Column(name = "billing_city"))
+    })
+    private Address billingAddress;
+
```

Comment:
Jak działają te @AttributeOverrides? Rozumiem, że apka ma "ignorować" zastane nazwy "streetName" i zamiast nich używać "billing_address" albo "delivery_address". Czy to zadziała w ten sposób, że potem stworzymy sobie encję Address, która będzie miała nazwy kolumn "streetName", "streeNumber" i tak dalej, no ale ta encja Address będzie tylko jedna a my będziemy chcieli mieć różne nazwy kolumn w zależności od tego czy to delivery czy billing address? Czy to jest kwestia w takim razie tylko "kosmetyczna" czy czegoś nie czaję?